docker/lib/dependabot/docker/utils/credentials_finder.rb

```diff
@@ -27,24 +26,15 @@ def credentials_for_registry(registry_hostname)
           build_aws_credentials(registry_details)
         end
 
-        # Return true if replaces-base is enabled else return false
-        def replaces_base?
-          return @registry_details["replaces-base"] == true if @registry_details&.key?("replaces-base")
-
-          false
-        end
-
-        def fetch_base_registry
-          return @registry_details["registry"] if @registry_details
+        def base_registry
+          @base_registry ||= credentials.find do |cred|
+            cred["type"] == "docker_registry" && cred["replaces-base"] == true
+          end
+          @base_registry ||= { "registry" => DEFAULT_DOCKER_HUB_REGISTRY, "credentials" => nil }
```

Comment:
@jakecoffman DEFAULT_DOCKER_HUB_REGISTRY is initialization is missing in credentials_finder.rb. It was used in two places in file_pasrser.rb. Can we make this common to both files

Reply:
Ah I missed that. That might be an indicator we could push more down into the credentials_finder.